### DISCUSSION TITLE
Can Fooocus Check Multiple Paths to Find Base Models when Switching Presets?

### DISCUSSION THREAD

**Discussion by DavidDragonsage** [lllyasviel/Fooocus#3490] [Q&A] [answered]
I run Fooocus from an NVME boot drive but I do not have enough space to store _all_ the base models there as well. Therefore I have placed only my two most frequently used base models on the NVME.

In config.txt I have this entry to allow the base model to be loaded from two paths:
"path_checkpoints": ["C:\\stable-diffusion-webui\\Fooocus\\models\\checkpoints", "E:\\stable-diffusion-webui\\models\\Stable-diffusion"],

This works fine when switching base models using the Models tab. Any models from either path can be loaded.

But when switching models using a Preset, Fooocus would start to download any model that was not in the NVME C: drive path.

This behavior was easily corrected by adding the  --disable-preset-download switch to the run.bat startup file. Now everything works correctly.

The only disadvantage is that Fooocus will no longer auto-update to any new base models or LoRAs when they are released in a new Fooocus version. 

I guess the problem is that Fooocus does not check both paths when changing presets. Is there a way to make it do so in config.txt?

I do not want to create my own Fooocus branch just to do this in coding because that creates a whole new nightmare of update chasing. Having the code up-to-date is much more important to me than having the models auto-updated.

**Comment by DavidDragonsage** [comment]
I have a workaround for this issue, and the only slight disadvantage is that models that are placed in the NVME boot drive will also be placed in the hard drive with all the rest of the models. To me, this duplication is not a problem at all - the hard drive has lots of room.

In config.txt, switch the order of the NVME boot drive and the hard drive paths. This ensures that auto-updates always go to the hard drive so that models that I use less often are not filling the NVME drive:
"path_checkpoints": ["E:\stable-diffusion-webui\models\Stable-diffusion", "C:\stable-diffusion-webui\Fooocus\models\checkpoints"],

Place the models on the NVME drive in a recognizable subfolder, for example "fast":
"C:\stable-diffusion-webui\Fooocus\models\checkpoints\fast"

In the presets for the special models, use the full path to the NVME drive, for example:
"default_model": "C:\\stable-diffusion-webui\\Fooocus\\models\\checkpoints\\fast\\SDXLFaetastic_v24.safetensors",

If you load or switch to this preset, the base model will be displayed with the full path under the Models tab, for example:
C:\stable-diffusion-webui\Fooocus\models\checkpoints\fast\SDXLFaetastic_v24.safetensors

But if you want to select this model from the Models tab without switching presets, be sure to select the model with "Fast\" in the path:
Fast\SDXLFaetastic_v24.safetensors
as shown in the attachment. You would avoid loading the copy of this model that is not prefixed by the word "Fast" because that is the one on the hard drive.

The auto-update switch can be restored to run.bat:  -always-download-new-model

The base models that are given the "fast" treatment will also exist on the hard drive - in fact they will be re-downloaded there if you delete them. In this example, that path is: E:\stable-diffusion-webui\models\Stable-diffusion

I am happy with this solution.

![SelectFastVersion](https://github.com/user-attachments/assets/90b17a7c-0b42-4eb5-a50c-e43606148745)

**Comment by mashb1t** [comment, marked as answer]
Referenced in https://github.com/lllyasviel/Fooocus/issues/3492 so i don't forget to check this. Thanks for reporting and your in-depth analysis.

**Comment by DavidDragonsage** [comment]
Well that's neat, it looks like you have already developed a fix for the underlying problem. That will simplify things considerably! 🙂
